src/main/java/edu/kis/powp/jobs2d/command/ImmutableCompoundCommand.java

```diff
@@ -12,9 +12,23 @@
     private List<DriverCommand> driverCommands = null;
 
     public ImmutableCompoundCommand(List<DriverCommand> driverCommands) {
+
         this.driverCommands = new ArrayList<>();
-        this.driverCommands.addAll(driverCommands);
+        for (Object object : driverCommands) {
+            if (object instanceof DefaultCompoundCommand) {
```

Comment:
object-oriented design makes often possible to avoid instanceof. This is the case.